### PR TITLE
Update CA1010 to consider IReadOnlyXXX as valid counterpart

### LINKED ISSUE #4676: CA1010 should not fire if readonly interface implemented
### Analyzer

**Diagnostic ID**: [CA1010](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1010)

### Describe the improvement

Sometimes, a type implements `IReadOnlyList<T>` instead of `IList<T>` to clarify readonly-ness. However, there are not readonly version for non-generic `IList`. CA1010 will fire for this case, and can be avoided.

### Describe suggestions on how to achieve the rule

When implementing readonly collection interfaces `IReadOnlyList<T>` and `IReadOnlyCollection<T>`, also treats the type satisfying the rule.

### PR DESCRIPTION
**Pull request by Evangelink**
Fix #4676

### CODE REVIEW

File: src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/CollectionsShouldImplementGenericInterfaceTests.cs [hunk 1]
```diff
@@ -15,37 +15,43 @@ namespace Microsoft.CodeQuality.Analyzers.ApiDesignGuidelines.UnitTests
 {
     public class CollectionsShouldImplementGenericInterfaceTests
     {
-        private static DiagnosticResult GetCA1010CSharpResultAt(int line, int column, string typeName, string interfaceName, string genericInterfaceName)
-#pragma warning disable RS0030 // Do not used banned APIs
```
Comment (Evangelink) [inline src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/CollectionsShouldImplementGenericInterfaceTests.cs:19]: I had to create the new overloads so I took a bit of extra and updated all other occurrences (that's one step forward for the no `RS0030`).

**Comment by Evangelink** [comment]
ping @mavasani

**Comment by mavasani** [comment]
Ping @Evangelink

**Comment by Evangelink** [comment]
There you go @mavasani, sorry for the delay!

**Comment by Evangelink** [comment]
It seems that there is a regression with rule `CA2255`.

**Comment by Evangelink** [comment]
@mavasani May I ask for help on the PR please?

**Pull request merged**